src/features/postsSlice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+export type PostsState = {
+  posts: Post[];
+  loading: boolean;
+  error: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loading: false,
+  error: false,
+};
+
+export const fetchPosts = createAsyncThunk(
+  'posts/fetchPosts',
+  (userId: number) => getUserPosts(userId),
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      return {
+        ...state,
+        posts: [],
+      };
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchPosts.pending, state => {
+      return { ...state, loading: false };
```

Comment:
There is a mistake in the 'pending' case. The 'loading' state should be set to 'true' when a fetch request is pending, not 'false'. This indicates that the data is currently being loaded.